### ISSUE TITLE
[PLL] Plain Language Lens

### ISSUE THREAD

**Issue by joofio** [Gravitate-Health/mvp-issues#5]
## **1. User Stories**

As a Patient, I want difficult medical terms to be highlighted so that I can click on them and get an easy-to-understand explanation.

## **2. Overview**
The system will apply a plain language lens to ePIs  to enhance readability and understandability and interactivity by highlighting complex terms/sentences and making them clickable objects for additional explanations.

## **2. Functional Requirements**

- The system should analyze sentences using readability scores (e.g., Flesch-Kincaid, Gunning Fog, SMOG).
- Sentences identified as difficult should be tagged for further enhancement.
- Tagged sentences should have an option to display a simplified version upon clicking.
- The system should use LLM/rule-base system to generate contextually accurate simplified sentences.

## **3. Technical Requirements**
- A preprocessing module should analyze and tag sentences for readability
- The readability score should be computed dynamically for each sentence or paragraph (the minimum possible).
- The tagged sentences should be assigned a CSS class to allow easy selection by the lens.
- The lens should collect sentences with the predefined tag and pass them through a processing system (e.g., LLM) to generate simplified versions.
- The clickable artifacts should be styled to distinguish them without disrupting document readability.
- The system should maintain accessibility compliance (e.g., WCAG standards).

**Comment by joofio** [comment]
meeting 5/12:

the idea right now is to add a summary of the information of the lens. only add  information due to legal reasons.

**Comment by joofio** [comment]
can we use the current summary to create (at least a draft of) this one? should be quick and easy.
@aalonsolopez ? what do you think?

**Comment by aalonsolopez** [comment]
yeah in fact this is almost the same as is working rn, only we have to change it from requesting a summary to just a language simplification

**Comment by joofio** [comment]
moved to tentative mvp3, or mvp4

**Comment by joofio** [comment]
Deploy and test.
preprocessor to be added like "readability scores" to tag difficult words
